lib/open-search-client.ts

```diff
@@ -0,0 +1,378 @@
+ * in this response should always be used in future next scroll requests.
+ */
+export const nextScroll = async (
+  prefixUrl: string,
+  scrollId: string,
+  scrollTimeout = 60
+): Promise<unknown> => gotEs(prefixUrl).post(
+  '_search/scroll',
+  {
+    json: {
+      scroll: `${scrollTimeout}m`,
+      scroll_id: scrollId,
+    },
+  }
+);
+
+/**
+ * Closes an open Open Search scroll query.
+ * See https://opensearch.org/docs/latest/opensearch/rest-api/scroll/
+ * for more details.
+ *
+ * @param prefixUrl - Open Search endpoint.
+ * @param scrollId - Scroll ID of the open scroll.
+ *
+ * @returns Open Search close scroll response.
+ */
+export const closeScroll = (
+  prefixUrl: string,
+  scrollId: string
+): Promise<CloseScrollResponse> => got4aws().delete(
+  `_search/scroll/${scrollId}`,
+  {
+    prefixUrl,
+    responseType: 'json',
+    resolveBodyOnly: true,
+  }
+);
+
+/**
+ * Deletes items from an Open Search index using the _bulk API.
+ * See: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html
+ *
+ * @param prefixUrl - Open Search endpoint.
+ * @param index - Name of the index where the items exist.
+ * @param ids - List of IDs to delete.
+ *
+ * @returns Result of Open Search bulk delete.
+ */
+export const bulkDelete = async (
+  prefixUrl: string,
+  index: string,
+  ids: string[]
+): Promise<CloseScrollResponse> => {
+  const bulkData = ids.map((id) => ({
+    delete: {
+      _index: index,
+      _type: '_doc',
+      _id: id,
+    },
+  }));
+
+  // eslint-disable-next-line no-return-await
+  return await got4aws().post(
+    '_bulk',
+    {
+      prefixUrl,
+      body: `${bulkData.map((d) => JSON.stringify(d)).join('\n')}\n`,
+      responseType: 'json',
+      resolveBodyOnly: true,
+    }
+  );
+};
+
+export interface PercolateDocumentFieldsOptions {
+  from?: number
+  size?: number
+}
+
+/**
+ * Executes a percolator query against the terms index using the given
+ * document index fields. This query takes document key/values and finds
+ * the terms that match those field values. It returns a modified
+ * percolator query result.
+ * See https://www.elastic.co/guide/en/elasticsearch/reference/6.8/query-dsl-percolate-query.html
+ * for more information.
+ *
+ * @param prefixUrl - Open Search endpoint.
+ * @param fields - Key/value index fields to use to match
+ * the percolator query.
+ * @param options - Options for the request.
+ *
+ * @returns Modified percolator query result.
+ * The result will be a list of objects that includes the matched term
+ * label, uri, and source_terminology.
+ */
+export const percolateDocumentFields = async (
+  prefixUrl: string,
+  fields: { title: string, contents: string },
+  options: PercolateDocumentFieldsOptions = {}
+): Promise<DocumentItemTerm[]> => {
+  const {
+    from = 0,
+    size = 300,
+  } = options;
+
+  const body = {
+    query: {
+      percolate: {
+        field: 'query',
+        document: {
```

Comment:
Why is this `...fields` and not `document: fields`?

Reply:
Heh, my brain just decided to write it that way.